tests/helpers/test_frame.py

```diff
@@ -92,3 +92,16 @@ async def test_prevent_flooding(caplog):
     assert what not in caplog.text
     assert key in frame._REPORTED_INTEGRATIONS
     assert len(frame._REPORTED_INTEGRATIONS) == 1
+
+
+async def test_prevent_flooding_missing_integration_frame(caplog):
+    """Test to ensure only one log when no integration is detected."""
```

Comment:
The test name and its docstring are misleading since there's nothing preventing repeating the same error message when the issue is not caused by an integration.